### PR TITLE
meade offset removed conditionally

### PR DESCRIPTION
**Pull request by anjok**
whole thing is explained here: https://github.com/OpenAstroTech/OpenAstroTracker-Firmware/issues/166

### CODE REVIEW

File: src/Longitude.cpp [hunk 1]
```diff
@@ -2,6 +2,17 @@
 #include "Utility.hpp"
 #include "Longitude.hpp"
 
+// define INDI in the local configuration if you want it to work
+#ifndef INDI
```
Comment (julianneswinoga) [inline src/Longitude.cpp]: Could you move this default setting to `Configuration_adv.hpp`? I think the `Misc Configuration` makes sense for this

File: src/Longitude.cpp [hunk 2]
```diff
@@ -2,6 +2,17 @@
...
+    #define INDI 0
```
Comment (julianneswinoga) [inline src/Longitude.cpp]: I would rename this `#define` to be a little more specific. `INDI_LONGITUDE_HACK` maybe?

**Review by julianneswinoga — CHANGES_REQUESTED** [review]
FYI in order for all the workflow checks to pass you also need to update `Changelog.md` and `Version.h`

File: src/Longitude.cpp (hunk 1 above)
Comment (anjok) [reply src/Longitude.cpp]: Sure

File: src/Longitude.cpp (hunk 2 above)
Comment (anjok) [reply src/Longitude.cpp]: Actually... it's the opposite, at least from my understanding. What it does is preserve the current parsing on Gg12 -> 180 - 12 for users of the current ASCOM driver, when the correct value should be - 12. The parsing assumes that KStars/Meade/ASCOM will have a 180 - x offset added when in fact it does not.

The whole thing is explained here (and I should referenced this in the subject): https://github.com/OpenAstroTech/OpenAstroTracker-Firmware/issues/166

I just named it INDI (which is a lame name, agreed, maybe INDI_SUPPORT) because I have a feeling there might be more stuff coming in to support KStars and I didn't want flags for all of them.

**Comment by anjok** [comment]
I don't get the needed change for changelog and version. Why is this on me to do that? How would I know which version this is going into? 

Normally, you'd merge some PRs to develop, then decide it's time for a release and do the changelog and version in a commit there...

**Comment by anjok** [comment]
Added version stuff, renamed and moved define  and force-pushed.

**Comment by julianneswinoga** [comment]
> I don't get the needed change for changelog and version. Why is this on me to do that? How would I know which version this is going into?
> 
> Normally, you'd merge some PRs to develop, then decide it's time for a release and do the changelog and version in a commit there...

1. It's up to the author how they would like to describe their changes in the changelog. It's not auto-generated from commits (which I'm assuming is what you are thinking about) because commits are directed more at developers, where the changelog is directed more at users.
2. For changes like this the next version is simply a minor version increment. We try to follow https://semver.org/ convention
3. For full releases the changes are thus very easy to create simply by looking at the prose changes in `Changelog.md` :smile:

**Comment by julianneswinoga** [comment]
> Added version stuff, renamed and moved define and force-pushed.

It would be appreciated if you didn't squash+force-push while a PR is under review (i.e. only do that right before the PR is going to be merged). It makes it very hard to follow incremental changes and review them accordingly.

File: Changelog.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+**V1.10.8 - Updates**
+- Fix Meade parsing from INDI clients.
+
 **V1.10.6 - Updates**
```
Comment (julianneswinoga) [inline Changelog.md:4]: It looks like you need to fetch and rebase your changes onto develop, you created your branch quite a while ago:
```bash
# Add an upstream URL
# This URL is for ssh, change it if you use HTTPS
git remote add upstream git@github.com:OpenAstroTech/OpenAstroTracker-Firmware.git
git fetch -p upstream
git rebase -i origin/develop # Interactive rebase
# Fix merge conflicts
git push -f # Force push is needed because the commits have changed
```

File: src/Longitude.cpp [hunk 4]
```diff
@@ -2,6 +2,12 @@
 #include "Utility.hpp"
 #include "Longitude.hpp"
 
+#if INDI_SUPPORT == 1
+    #define LONGITUDE_OFFSET 0L
```
Comment (julianneswinoga) [inline src/Longitude.cpp:6]: Could you also add in
```C++
#pragma message("Note: INDI offset used, firmware will not function correctly with ASCOM!")
```
? Just so that we can have traceability if a user is having problems.

File: src/Longitude.cpp (hunk 2 above)
Comment (julianneswinoga) [reply src/Longitude.cpp]: We can re-evaluate at a later date if more INDI-specific changes are needed. More specific is better in the current case.

File: Configuration_adv.hpp [hunk 5]
```diff
@@ -569,3 +569,9 @@
         #error "Debugging not supported on this platform"
     #endif
 #endif
+
+// define INDI_SUPPORT in the local configuration if you want it to work
+// currently this will fix Meade longitude parsing
+#ifndef INDI_SUPPORT
```
Comment (julianneswinoga) [inline Configuration_adv.hpp:575]: Please move this to the `Misc Configuration` section of this file

**Review by julianneswinoga — CHANGES_REQUESTED** [review]

**Comment by JakobPetersson** [comment]
@cameronswinoga Should I create a new PR with these changes if i wish to see them merged?

**Comment by anjok** [comment]
You also need to fix the sign of the utc offset in the meadeparser

**Comment by anjok** [comment]
Like so, but you also need to move the flag to Configuration_avd.hpp (or just skip these conditionals alltogether and just fix it, as this is only about windows users not needing to update the ASCOM driver) : 
``` /////////////////////////////////
 void Mount::setLocalUtcOffset(int offset)
 {
+    #if INDI
+    offset = -offset;
+    #endif
     _localUtcOffset = offset;
```

**Comment by JakobPetersson** [comment]
Fixes for this are included in #181.
I suggest we close this and move forward there.

**Comment by ClutchplateDude** [comment]
Fixes made in #181
